engine/include/Engine/Intermediate/Component.hpp

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (C) 2021 Amara Sami, Dallard Thomas, Nardone William, Six Jonathan
+ * This file is subject to the LGNU license terms in the LICENSE file
+ *	found in the top-level directory of this distribution.
+ */
+
+#pragma once
+
+namespace GPE
+{
+class GameObject;
+
+class Component
+{
+protected:
+    GameObject& m_gameObject;
+    bool        m_isActivated{true};
+
+public:
+    inline Component(GameObject& owner) noexcept;
+    inline Component() noexcept                       = delete;
+    inline Component(const Component& other) noexcept = delete;
+    inline Component(Component&& other) noexcept      = default;
+    inline virtual ~Component() noexcept              = default;
+    inline Component& operator=(const Component& other) noexcept = delete;
+    inline Component& operator=(Component&& other) noexcept = default;
+
+    [[nodiscard]] constexpr inline GameObject& getGameObject() noexcept;
+
+    [[nodiscard]] constexpr inline const GameObject& getGameObject() const noexcept;
```

Comment:
**Suggestion**

I would rename the `getGameObject()` methods `owner()` or `getOwner()`. I find it conveys more meaning, and is shorter